tests/integration/src/test/resources/org/mule/test/integration/routing/choice-router-config.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns="http://www.mulesoft.org/schema/mule/core" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:spring="http://www.springframework.org/schema/beans"
+      xmlns:test="http://www.mulesoft.org/schema/mule/test"
+      xsi:schemaLocation="
+               http://www.mulesoft.org/schema/mule/test http://www.mulesoft.org/schema/mule/test/current/mule-test.xsd
+               http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-current.xsd
+               http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd">
+
+    <flow name="flow">
+        <choice>
+            <when expression="#[payload == 'A']">
```

Comment:
Does not make more sense to implement the proposal defined at https://www.mulesoft.org/jira/browse/MULE-6129?focusedCommentId=80558&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-80558 instead of supporting a choice with not default?